spec/lib/abstractivator/tree_visitor_spec.rb

```diff
@@ -243,6 +249,29 @@ def self.example(description, values)
               result: [{path: 'c/0', tree: 2, mask: 1}]
     end
 
+    context 'when comparing sets' do
+      example 'handles empty trees',
+              tree:   {set: build_set(0)},
+              mask:   {set: Abstractivator::SetMask.new(build_set(1), ->(x) { x[:id] })},
```

Comment:
I guess the downside to this example stuff is, since the `example` calls are class level, all the normal rspec `let!` and `before :each` stuff doesn't work :/